frontend/ui/widget/filechooser.lua

```diff
@@ -234,6 +234,13 @@ local FileChooser = Menu:extend{
                 if item.opened then
                     local doc_settings = DocSettings:open(item.path)
                     percent_finished = doc_settings:readSetting("percent_finished")
+                    local summary = doc_settings:readSetting("summary")
+
+                    -- books marked as "finished" should be considered the same as 100%
+                    if summary and summary.status == "complete" then
```

Comment:
is this more correct? i am not really experienced in lua, does it maybe work similar to js's `&&` where `let t = a && b`  if `a` is truthy results in `t` getting the value of `b` (`t === b`) otherwise being `false` (or maybe in this case being `nil`)?

Reply:
wait, i forgot i can simply test this, and yes, it is the same behavior as in js, so this could be a variable; should it?

Reply:
The more obvious parallel with the conditional (`if (summary && summary.status) {}` comes down to the same thing.[1] I.e., if truthy, continue to the right, if also truthy, conditional as a whole true.

Assignment doesn't work the same: you'd have to write something like `local var = condition and true or false` if you wanted that specific outcome. Otherwise it's more akin to `let a = a ? a : b`.

[1] You could use `if (summary?.status)` these days, though I'm not quite convinced that optional chaining isn't an antipattern.

Reply:
The condition is good.
You can move `percent_finished` (line 236) after this.

Reply:
> wait, i forgot i can simply test this, and yes, it is the same behavior as in js, so this could be a variable; should it?

GH hadn't shown me this new reply while I was writing mine. I only just saw it via email. I would consider doing that undesirable, reducing clarity for no apparent reason. Clarity could potentially be improved that way when something else was also involved, but at least in this codebase we prefer to address that differently, for example along these lines:

```lua
if
  summary and summary.status
  and cover and cover.x and cover.y
  and hello and hello.goodbye
then
```

You could also write that as something like `summary_status and cover_x and cover_y and hello_goodbye` defined above but that's just not what we do here. ;-)

Though unlike using spaces instead of tabs, I also prefer it that way.

Reply:
> The condition is good. You can move `percent_finished` (line 236) after this.

Grr, nor this reply. I thought GH updated the comments more or less live. Oh well.

Reply:
i will mark this as resolved with "leave it as-is" and moving the "percent_finished" assignment after the early-return (done with c190668d6fe8b04d3773bcf6c7183f3a475ae196)